### PR TITLE
Adding function listenAndServe to port 8000

### PR DESCRIPTION
**Pull request by polipopoliko**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,16 +3,21 @@ package main
 import (
 	"fmt"
 	"log"
+	"net/http"
 
-	_ "github.com/vicimilenia/tsaving/database"
+	"github.com/polipopoliko/tsaving/tsaving/api"
+	"github.com/polipopoliko/tsaving/tsaving/database"
```
Comment (vicimilenia) [inline main.go]: change /polipopoliko/ to /ndv6/ in line 8 and  9
```suggestion
	"github.com/ndv6/tsaving/database"
```

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// change the module path to your directory
+module github.com/polipopoliko/tsaving/tsaving
```
Comment (vicimilenia) [inline go.mod]:
```suggestion
module github.com/ndv6/tsaving
```

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// change the module path to your directory
```
Comment (vicimilenia) [inline go.mod]: No need for this comment, as ideally all repository cloned to personal laptops follows the format github.com/ndv6/tsaving

**Review by david1312 — APPROVED** [review]

File: api/router.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+func Router(db *sql.DB) *chi.Mux {
+	chiRouter := chi.NewRouter()
+	chiRouter.Get("/", HomeHandler)
+	chiRouter.NotFound(NotFoundHandler)
+	return chiRouter
+}
+
+func HomeHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (vicimilenia) [inline api/router.go]: Move this to another file named home or homepage or homehandler under the api folder

File: api/router.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	fmt.Fprintln(w, "Welcome to Tsaving")
+}
+
+func NotFoundHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (vicimilenia) [inline api/router.go]: Move this to another file named not_found or not_found_page or not_found_handler.go under the api folder

**Review by vicimilenia — CHANGES_REQUESTED** [review]

**Review by celynolivia — APPROVED** [review]

File: api/router.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package api
+
+import (
+	"database/sql"
+
+	"github.com/polipopoliko/ndv6/tsaving/api/home/homepage"
+	"github.com/polipopoliko/ndv6/tsaving/api/not_found/not_found_page"
```
Comment (vicimilenia) [inline api/router.go]:
```suggestion
	"github.com/ndv6/tsaving/api/home/homepage"
	"github.com/ndv6/tsaving/api/not_found/not_found_page"
```

File: go.mod [hunk 7]
```diff
@@ -0,0 +1,10 @@
+module github.com/polipopoliko/ndv6/tsaving
```
Comment (vicimilenia) [inline go.mod]:
```suggestion
module github.com/ndv6/tsaving
```

File: main.go [hunk 8]
```diff
@@ -3,16 +3,21 @@ package main
 import (
 	"fmt"
 	"log"
+	"net/http"
 
-	_ "github.com/vicimilenia/tsaving/database"
+	"github.com/polipopoliko/ndv6/tsaving/api"
+	"github.com/polipopoliko/ndv6/tsaving/database"
```
Comment (vicimilenia) [inline main.go]:
```suggestion
	"github.com/ndv6/tsaving/api"
	"github.com/ndv6/tsaving/database"
```

**Review by vicimilenia — CHANGES_REQUESTED** [review]

**Review by vicimilenia — APPROVED** [review]

**Pull request merged**